bentoml/server/api_server.py

```diff
@@ -443,44 +454,75 @@ def handle_ml_processing(self):
         URL_SUBSCRIPTION = self.ngsild_cb_url + '/ngsi-ld/v1/subscriptions/'
         SUBSCRIPTION_INPUT_DATA = 'urn:ngsi-ld:Subscription:input:data:'+str(uuid.uuid4())
         AT_CONTEXT = [ self.ngsild_at_context ]
+        ENTITY_INPUT_TYPE = self.ngsild_ml_model_entity_input_type
+        ATTRIBUTE_INPUT_DATA = self.ngsild_ml_model_input.split(',')
+        TIME_INTERVAL = self.ngsild_ml_model_time_interval
+        logger.info('ATTRIBUTE_INPUT_DATA: %s\n', ATTRIBUTE_INPUT_DATA)
 
         # Get the POST data
         mlprocessing_notification = request.get_json()
-
-        # Get the MLProcessing entity and extract the entities information (to be extended later)
-        mlprocessing_entity = mlprocessing_notification['data'][0]
-        logger.info("Got following MLProcessing entity: %s", mlprocessing_entity)
-
-        input_entity_id = mlprocessing_entity['entities'][0]['id']
-        input_entity_type = mlprocessing_entity['entities'][0]['type']
-
-        ATTRIBUTE_INPUT_DATA = self.ngsild_ml_model_input
-
-        # We use the content of the MLProcessing only to get the entity id and type. 
-        # The input attribute(s) is(are) part of the the MLModel definition.
-        json_ = {
-            '@context': AT_CONTEXT,
-            'id': SUBSCRIPTION_INPUT_DATA,
-            'type': 'Subscription',
-            'entities': [
-                {
-                    'id': input_entity_id,
-                    'type': input_entity_type
+        logger.info('Notification received: %s', mlprocessing_notification)
+
+        #### 
+
+        # Getting the EntityID where to get input data
+        ENTITY_INPUT_DATA = mlprocessing_notification['data'][0]['entityID']['value']
+        logger.info('ENTITY_INPUT_DATA: %s', ENTITY_INPUT_DATA)
+
+        # Only for testing with postman mock server, replace
+        # 'uri': request.url_root + '/ngsi-ld/ml/predict'
+        # by postman mock server id
+        # 'uri': 'https://0ba2eb3a-2ff5-4a72-9a6f-f430f9f41ad3.mock.pstmn.io/ngsi-ld/ml/predict' 
+
+        # When entityID value is 'TIME', we set a TIME subscription
+        # Else a subscription based on change of attributes
+        if ENTITY_INPUT_DATA == 'TIME':
+            json_ = {
+                '@context': AT_CONTEXT,
+                'id': SUBSCRIPTION_INPUT_DATA,
+                'type': 'Subscription',
+                'entities': [
+                    {
+                        'id': ENTITY_INPUT_DATA,
```

Comment:
as per line 479, `ENTITY_INPUT_DATA` is equal to `TIME`, this will never notify...